QModManager/SummaryLogger.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace QModManager
+{
+    using System;
+    using System.Collections.Generic;
+    using QModManager.API.ModLoading;
+    using QModManager.API.ModLoading.Internal;
+    using QModManager.DataStructures;
+    using Logger = Utility.Logger;
+
+    internal static class SummaryLogger
+    {
+        internal static void LogSummaries(PairedList<QMod, ModStatus> mods)
+        {
+            CheckOldHarmony(mods.Keys);
+            LogStatus(mods, ModStatus.CanceledByAuthor, "The following mods have been disabled internally by the mod author", Logger.Level.Info);
+            LogStatus(mods, ModStatus.CanceledByUser, "The following mods have been disabled by user configuration", Logger.Level.Info);
+            LogStatus(mods, ModStatus.PatchMethodFailed, "The following mods failed during patching", Logger.Level.Error);
+            LogStatus(mods, ModStatus.TooManyPatchMethods, "The following mods had too many patch methods and were canceled", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.MissingPatchMethod, "The following mods had no patch methods to run", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.CircularLoadOrder, "The following mods could not be loaded due to circular load order", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.CircularDependency, "The following mods could not be loaded due to circular dependencies", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.MissingDependency, "The following mods could not be loaded due to missing dependencies", Logger.Level.Warn);
```

Comment:
Everything is here should use `Logger.Level.Error`

Reply:
I've used `Warn` for all issues that were caught _before_ we started any patching.  
These shouldn't cause any major problems in-game since the mods were never loaded.  

I used `Error` for the issues that come up during the patching which could leave the game in a half-patched state.

Reply:
From an user's perspective, mods that failed to load are "errored"